modules/calc/src/main/java/com/opengamma/strata/calc/marketdata/CalculationMarketData.java

```diff
@@ -27,30 +27,44 @@
 public interface CalculationMarketData {
 
   /**
-   * Returns the valuation dates of the scenarios, one for each scenario.
+   * Gets the valuation dates of the scenarios, one for each scenario.
    *
    * @return the valuation dates of the scenarios, one for each scenario
    */
   public abstract MarketDataBox<LocalDate> getValuationDate();
 
   /**
-   * Returns the number of scenarios.
+   * Gets the number of scenarios.
    *
    * @return the number of scenarios
    */
   public abstract int getScenarioCount();
 
+  //-------------------------------------------------------------------------
   /**
-   * Returns a box that can provide an item of market data for a scenario.
+   * Checks if this set of data contains a value for the specified key.
    *
-   * @param <T>  type of the market data
-   * @param key  a key identifying the market data
-   * @return a list of market data values, one from each scenario
+   * @param <T>  the type of the market data
+   * @param key  the key identifying the item of market data
+   * @return true if this set of data contains a value for the specified key
+   */
+  public abstract <T> boolean containsValue(MarketDataKey<T> key);
+
+  /**
+   * Gets a box that can provide an item of market data for a scenario.
+   * <p>
+   * The market data is valid for the valuation date.
```

Comment:
This line doesn't really mean anything any more. There are potentially multiple valuation dates and multiple values for the market data. The market data is valid for the valuation date which is looked up using the same index. But that is too much of a mouthful to be a useful comment.